### PR TITLE
fix variable length iv issue in gcm

### LINKED ISSUE #53: AES-GCM-256 with 16 zero byte IV.
An Apple Wallet product (as much as I can say due to NDA) uses aes-gcm-256. The 32 byte key is created using a kdf with ECDH derived key material. As the keys are only ever used once, Apple is using a 16 byte IV with all zero bytes. 

We developed an implementation of the Apple crypto protocol that works using the `crypto` module, but seems to fail with browserify-aes, hitting an assertion of IV length == 12. When commenting out that assertion, it complains about a bad authentication state/tag (or the like)

I'm not really up to speed with AES/GCM so not sure where to start with this, but I could help out with some failing tests if that would be of assistance.

### PR DESCRIPTION
**Pull request by calvinmetcalf**
fixes #53

### CODE REVIEW

File: authCipher.js [hunk 1]
```diff
@@ -17,13 +18,39 @@ function xorTest (a, b) {
   return out
 }
 
+function calcIv (self, iv, ck) {
```
Comment (dcousens) [inline authCipher.js]: Do you have a source for where this algorithm came from?

Comment (calvinmetcalf) [reply authCipher.js]: primarily https://csrc.nist.gov/publications/detail/sp/800-38d/final page 15

**Comment by sublimator** [comment]
Nice!

File: authCipher.js (hunk 1 above)
Comment (dcousens) [reply authCipher.js]: partially verifed,  couldn't find the `[..., 2]` ref tho?

**Comment by sublimator** [comment]
I was testing this today, and it seems this only works some of the times. It might pay to do some fuzzing ?

It seems to be creating the wrong tag? I get an error at https://github.com/crypto-browserify/browserify-aes/blob/70131287cc603cda31d32c5ee66c3a9da2b936c4/authCipher.js#L92

**Comment by sublimator** [comment]
I can try and log the key/tag/data etc when it fails if that would help

**Comment by sublimator** [comment]
Nevermind, it seems to be a bug in our code elsewhere. I checked the secret/encrypted/tag against another impl, and got a similar error

**Pull request merged**